### PR TITLE
Fix for Dedicated Networking

### PR DESCRIPTION
**Pull request by joshuajerin**
- Allow IP Allow List to work with Dedicated Networking
- Creating 2 more hostname if Dedicated Networking is enabled (1 if DN enabled, 1 if Standby is enabled) to not disrupt the original one

### CODE REVIEW

File: tembo-operator/src/dedicated_networking.rs [hunk 1]
```diff
@@ -356,10 +356,16 @@ async fn reconcile_dedicated_networking_service(
         service_name, namespace, service_type, lb_scheme
     );
 
+    let external_dns_hostname = if is_standby {
+        format!("dedicated-ro.{}.{}", namespace, basedomain)
+    } else {
+        format!("dedicated.{}.{}", namespace, basedomain)
+    };
```
Comment (vrmiguel) [inline tembo-operator/src/dedicated_networking.rs:363]: Alternatively you could do something like this:

```rust
    let external_dns_hostname = format!(
        "dedicated{prefix}.{namespace}.{basedomain}",
        prefix = if is_standby { "-ro" } else { "" }
    );
```

Whichever you prefer

Comment (joshuajerin) [reply tembo-operator/src/dedicated_networking.rs]: Oh nice! Thank you

**Review by vrmiguel — APPROVED** [review]

File: tembo-operator/tests/integration_tests.rs [hunk 2]
```diff
@@ -2168,7 +2168,7 @@ mod test {
             .as_ref()
             .expect("Annotations should be present");
         let basedomain = std::env::var("DATA_PLANE_BASEDOMAIN").unwrap();
-        let expected_hostname = format!("{}.{}", namespace, basedomain);
+        let expected_hostname = format!("dedicated.{}.{}", namespace, basedomain);
```
Comment (ianstanton) [inline tembo-operator/tests/integration_tests.rs:2171]: Let's add one for when `is_standby == true`, where `expected_hostname` would equal `format!("dedicated-ro.{}.{}", namespace, basedomain)`

**Review by ianstanton — CHANGES_REQUESTED** [review]
Just a few changes requested! Can you also add some details to the description?

**Comment by ianstanton** [comment]
I think we have some flaky tests in the `tembo-operator` functional tests. I have seen this one failing in my branches as well: `test::functional_test_restart_and_update_replicas`

**Review by ianstanton — APPROVED** [review]

**Pull request merged**